### PR TITLE
Camel- to snake case

### PR DESCRIPTION
**Pull request by marcus-oscarsson**
Hi all,

This PR is based on our updates for the Beamline object pr #417, the first batch of commits will removed once accepted.

As we discussed during our last meeting, we have started with the camel to snake case conversion. We chose get_value, set_value and get_positions and we changed it everywhere even in site specific folders. As usual let us know if there's a commit that you don't like and we will remove it.

We though that we would do this in batches of a few methods per time so that the changes remain reasonably manageable.

What do you think,
Marcus and Antonia

### CODE REVIEW

File: HardwareObjects/Resolution.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def getState(self):
 
     def connectNotify(self, signal):
         if signal == "stateChanged":
-            self.dtoxStateChanged(HWR.beamline.detector.detector_distance.getState())
+            self.dtoxStateChanged( HWR.beamline.detector.detector_distance.getState())
```
Comment (rhfogh) [inline HardwareObjects/Resolution.py]: Remove extra space?

File: HardwareObjects/Resolution.py [hunk 2]
```diff
@@ -175,7 +175,7 @@ def move(self, pos, wait=False):
         if wait:
             HWR.beamline.detector.detector_distance.syncMove(self.res2dist(pos))
         else:
-            HWR.beamline.detector.detector_distance.move(self.res2dist(pos))
+             HWR.beamline.detector.detector_distance.move(self.res2dist(pos))
```
Comment (rhfogh) [inline HardwareObjects/Resolution.py]: Remove extra space?

File: HardwareObjects/Transmission.py [hunk 3]
```diff
@@ -11,7 +11,7 @@ def __init__(self, name):
         self.indexes = []
         self.attno = 0
         # TO DO: clean this!!!
-        self.getValue = self.get_value
+        self.get_value = self.get_value
```
Comment (rhfogh) [inline HardwareObjects/Transmission.py]: Remove this one (and search for others?) 

ALso, is this the time ot get rid of e.g. getA
ttFactor??

File: HardwareObjects/XRFSpectrum.py [hunk 4]
```diff
@@ -237,15 +237,15 @@ def spectrumCommandFinished(self, result):
         self.scanning = False
         if result is not False:
             try:
-                mcaData = self.getChannelObject("mca_data").getValue()
-                mcaCalib = self.getChannelObject("calib_data").getValue()
+                mcaData = self.getChannelObject("mca_data").get_value()
```
Comment (rhfogh) [inline HardwareObjects/XRFSpectrum.py]: You are sure that ChannelObjects should also have tgheir functions renamed? It is not externally defined - and so will still be getValue???

File: HardwareObjects/XSDataAutoprocv1_0.py [hunk 5]
```diff
@@ -787,7 +787,7 @@ def marshal(self):
         oStreamString = StringIO()
         oStreamString.write(unicode('<?xml version="1.0" ?>\n'))
         self.export(oStreamString, 0, name_="XSDataXdsCompletenessEntry")
-        oStringXML = oStreamString.getvalue()
+        oStringXML = oStreamString.get_value()
```
Comment (rhfogh) [inline HardwareObjects/XSDataAutoprocv1_0.py]: Again, this is a standard python library function. According to the documentation it has 'getvalue' (all lower case, no spacing). I think you might need to rethink your global replaces (case sensitivity for this one), also for the CHannelObjhects (see above). There may be others.

File: HardwareObjects/XSDataCommon.py [hunk 6]
```diff
@@ -169,7 +169,7 @@ def marshal(self):
         oStreamString = StringIO()
         oStreamString.write(unicode('<?xml version="1.0" ?>\n'))
         self.export(oStreamString, 0, name_="XSConfiguration")
-        oStringXML = oStreamString.getvalue()
+        oStringXML = oStreamString.get_value()
```
Comment (rhfogh) [inline HardwareObjects/XSDataCommon.py]: See comment abopve - PYthon library function getvalue

File: HardwareObjects/abstract/AbstractVideoDevice.py [hunk 7]
```diff
@@ -223,7 +223,7 @@ def get_jpg_image(self):
 
             strbuf = StringIO()
             image.save(strbuf, "JPEG")
-            jpgimg_str = strbuf.getvalue()
+            jpgimg_str = strbuf.get_value()
```
Comment (rhfogh) [inline HardwareObjects/abstract/AbstractVideoDevice.py]: Also suspected problem - standard python object probably does not have get_value

File: HardwareObjects/mockup/CatsMaintMockup.py [hunk 8]
```diff
@@ -366,7 +366,7 @@ def _executeServerTask(self, method, *args):
         # after launching a transfer
         # after setting refresh in the Tango DS to 0.1 s a wait of 1s is enough
         time.sleep(1.0)
-        while str(self._chnPathRunning.getValue()).lower() == "true":
+        while str(self._chnPathRunning.get_value()).lower() == "true":
```
Comment (rhfogh) [inline HardwareObjects/mockup/CatsMaintMockup.py]: Channel object - double check

File: HardwareObjects/mockup/DataAnalysisMockup.py [hunk 9]
```diff
@@ -39,7 +39,7 @@ def get_html_report(self, edna_result):
         html_report = None
 
         try:
-            html_report = str(edna_result.getHtmlPage().getPath().getValue())
+            html_report = str(edna_result.getHtmlPage().getPath().get_value())
```
Comment (rhfogh) [inline HardwareObjects/mockup/DataAnalysisMockup.py]: Double check?

File: HardwareObjects/mockup/TransmissionMockup.py [hunk 10]
```diff
@@ -8,7 +8,7 @@ def __init__(self, name):
         self.labels = []
         self.bits = []
         self.attno = 0
-        self.getValue = self.get_value
+        self.get_value = self.get_value
```
Comment (rhfogh) [inline HardwareObjects/mockup/TransmissionMockup.py]: Remove this one? There may be others

File: HardwareObjects/queue_entry.py [hunk 11]
```diff
@@ -1540,7 +1540,7 @@ def pre_execute(self):
         xray_centering = self.get_data_model()
         reference_image_collection = xray_centering.reference_image_collection
         reference_image_collection.grid = (
-            HWR.beamline.graphics.create_auto_grid()
+            HWR.beamline.microscope.shapes.reate_auto_grid()
```
Comment (rhfogh) [inline HardwareObjects/queue_entry.py]: Typo

File: HardwareObjects/queue_model_objects.py [hunk 12]
```diff
@@ -2131,21 +2131,21 @@ def dc_from_edna_output(
 
             try:
                 acquisition_parameters.osc_start = (
-                    goniostat.getRotationAxisStart().getValue()
+                    goniostat.getRotationAxisStart().get_value()
```
Comment (rhfogh) [inline HardwareObjects/queue_model_objects.py]: Presumably you have double checked that these EDNA objects also ahve a get_value function, or a getting one.

Comment (marcus-oscarsson) [reply HardwareObjects/queue_model_objects.py]: Nope we have not, that has to be changed back, ill go trough those tomorrow to see if I can find a good solution for those

File: HardwareObjects/Resolution.py (hunk 1 above)
Comment (marcus-oscarsson) [reply HardwareObjects/Resolution.py]: already done in previous PR, your commenting thins that belongs to #417 that have already been fixed

File: HardwareObjects/Transmission.py (hunk 3 above)
Comment (marcus-oscarsson) [reply HardwareObjects/Transmission.py]: Yes this mapping should be removed

getAttFactor ?, maybe it is but what has it to do with this ?

File: HardwareObjects/XRFSpectrum.py (hunk 4 above)
Comment (marcus-oscarsson) [reply HardwareObjects/XRFSpectrum.py]: Yes they already have a get_value defined as far as I know, its one of the inconsistent usages

File: HardwareObjects/Resolution.py (hunk 1 above)
Comment (rhfogh) [reply HardwareObjects/Resolution.py]: Sorry - it was in the text I was looking at, and it was quicker to comment and move on than to check whether I had come across it before and whether I should or should not recomment.

File: HardwareObjects/XRFSpectrum.py (hunk 4 above)
Comment (rhfogh) [reply HardwareObjects/XRFSpectrum.py]: Cool

File: HardwareObjects/XSDataAutoprocv1_0.py (hunk 5 above)
Comment (marcus-oscarsson) [reply HardwareObjects/XSDataAutoprocv1_0.py]: Yes, you are right I need to take a look at that one, I thought I did a case sensitive replace

File: HardwareObjects/Transmission.py (hunk 3 above)
Comment (rhfogh) [reply HardwareObjects/Transmission.py]: getAttFactor = get_value is the next line, in hte file, so it sort of suggested itself. 
But I was unsure whether this was the right time to make the change - I agree with your general approach to change a few at a time, so, ineed, let us leave it till later.

File: HardwareObjects/queue_entry.py (hunk 11 above)
Comment (marcus-oscarsson) [reply HardwareObjects/queue_entry.py]: This is not really a typo, this was done intentionally

Comment (rhfogh) [reply HardwareObjects/queue_entry.py]: 'reate' instead of 'create'? Well, if you say so.

**Comment by marcus-oscarsson** [comment]
Yes, Ill check the case sensitivity, sorry about that I though it was already the case

File: HardwareObjects/queue_entry.py (hunk 11 above)
Comment (marcus-oscarsson) [reply HardwareObjects/queue_entry.py]: oh, sorry yes that type :), yes that is really a typo :)

File: HardwareObjects/Resolution.py (hunk 1 above)
Comment (marcus-oscarsson) [reply HardwareObjects/Resolution.py]: :), i understand that ...

File: HardwareObjects/queue_entry.py (hunk 11 above)
Comment (marcus-oscarsson) [reply HardwareObjects/queue_entry.py]: Ok, taken care of in #417

**Comment by marcus-oscarsson** [comment]
So I looked through the setValue and getValue usage and it seems like the inconsistency is only for the Exporter protocol were the methods set_value and get_value were added at some point. The other protocols are all still using setValue and getValue and the rest of the API's looks very consistent and are all camel case.

As the call's to warn with a deprecation warning was added I thought that it was exporter that was the exception and all the rest already basically were snake cased. So it makes me question if we should as @rhfogh suggested in a comment above at all change the Commands and Channel API.

Is it maybe enough to focus the camel to snake case efforts on HardwareObjects ?

I've removed the set and getValue commits so that we can discuss this a bit further. I've only kept the commits for get_positions which is on Diffractometer and hopefully less drastic.

**Comment by rhfogh** [comment]
Nine days passed, much discussion, no objections. Time to merge

**Pull request merged**